src/frontend/pages/products/Filters.jsx

```diff
@@ -0,0 +1,126 @@
+import React from "react";
+import "./products.css";
+import { useFilter } from "../../contexts";
+import { filterAction } from "../../reducer";
+
+function Filters() {
+	const { state, dispatch } = useFilter();
+
+	const { sortByPrice, priceRange, filterByCategory, ratings } = state;
+
+	const categories = [
+		"childrenBooks",
+		"fiction",
+		"newReleases",
+		"selfHelp",
+		"biographies",
+	];
+
+	const ratingsValue = [4, 3, 2, 1];
+
+	return (
+		<div>
+			<aside className="aside__left">
+				<div className="aside__filter">
+					<h1 className="filter__heading">Filter</h1>
+					<h3 onClick={() => dispatch({ type: "Clear" })}>Clear</h3>
+				</div>
+				<h1 className="filter__heading">Price</h1>
+				<div>
+					<div className="price__range">
+						<p>100</p>
+						<p>500</p>
+						<p>1000</p>
+					</div>
+					<input
+						value={priceRange || 100}
+						type="range"
+						min="100"
+						max="1000"
+						onChange={(e) =>
+							dispatch({
+								type: filterAction.PRICE_RANGE,
+								payload: e.target.value,
+							})
+						}
+					/>
+				</div>
+
+				<h1 className="filter__heading">Category</h1>
+				<ul className="aside__list">
+					{categories.map((category) => (
+						<li key={category}>
+							<input
+								checked={filterByCategory[category]}
+								type="checkbox"
+								id={category}
+								onChange={() =>
+									dispatch({
+										type: filterAction.FILTER_BY_CATEGORY,
+										payload: category,
+									})
+								}
+							/>
+							<label>{category}</label>
```

Comment:
we can associate `label` with `input` element using id  or we can just wrap input inside the label element

Reply:
yes done